src/main/java/org/openrewrite/java/dependencies/DependencyVulnerabilityCheck.java

```diff
@@ -174,7 +174,8 @@ public Collection<SourceFile> generate(Accumulator acc, ExecutionContext ctx) {
             for (MinimumDepthVulnerability vDepth : vulnerabilitiesByGav.getValue()) {
                 Vulnerability v = vDepth.getVulnerability();
                 ResolvedGroupArtifactVersion gav = vulnerabilitiesByGav.getKey();
-                boolean fixWithVersionUpdateOnly = new LatestPatch(null).isValid(gav.getVersion(), v.getFixedVersion());
+                boolean fixWithVersionUpdateOnly = (vDepth.getMinDepth() == 0 || Boolean.TRUE.equals(overrideManagedVersion))
+                        && new LatestPatch(null).isValid(gav.getVersion(), v.getFixedVersion());
```

Comment:
This is the relevant change; the rest is formatting changes applied as part of our effort to standardize.

Reply:
I suppose that you have checked that in these cases the depth is 0 instead of 1

Reply:
Yes I downloaded a CSV from the platform where there was a mix of depths; with zero in there for direct dependencies.